.style-dictionary/build.js

```diff
@@ -92,7 +92,22 @@ StyleDictionary.registerFormat({
     dictionary.allProperties.forEach((prop) => {
       tokens[prop.name] = prop;
     });
-    return `module.exports = ${JSON.stringify(tokens, null, '  ')}`;
+    return `export default ${JSON.stringify(tokens, null, '  ')}`;
+  },
+});
+
+/**
+ * Custom Format: JS ESM
+ * This custom format is based on `javascript/module` but it uses `export default` instead of `module.exports`
+ */
```

Comment:
Just FYI there is also the [`es6` format](https://amzn.github.io/style-dictionary/#/formats?id=javascriptes6)

This format returns each individual property as an export instead of exporting an object, so it may not work  here.

Reply:
Yeah I tried that and it wasn't working because lots of the patterns rely on nested objects/destructuring and the es6 format flattened them